src/admin_extra/service_admin_extra_roster.erl

```diff
@@ -352,20 +352,21 @@ build_list_users(Group, [{User, Server}|Users], Res) ->
 -spec push_roster_item(jid:luser(), jid:lserver(), jid:user(),
         jid:server(), Action :: push_action()) -> 'ok'.
 push_roster_item(LU, LS, U, S, Action) ->
+    JID = jid:make(LU, LS, <<>>),
```

Comment:
```suggestion
    JID = jid:make_noprep(LU, LS, <<>>),
```

Reply:
Just because it says so in the spec is not enough for me to trust it's true. I cannot be sure from all the callers of this function that the input is already stringprepped. I see that this function is called from a command, which is external input, which makes me think that an admin might just make a mistake when giving it as input, so I prefer to play safe. Worst thing that can happen is, that it will be just as it was before, just that instead of `ejabberd_sm` doing the stringprep, it is now the command itself.

Reply:
Sounds good, you're right. I've not looked at this function's callers.